lib/src/pgn.dart

```diff
@@ -0,0 +1,68 @@
+import './models.dart';
+import "./position.dart";
+
+// Comment: Could we include a parent variable?
+// Comment: Do we want to separate ChildNode and Node?
+//   The idea here is that data becomes T? data
+// Comment: Is a separate walk function required if we have the transform function?
```

Comment:
A separate transform and walk function is like a separate map and forEach function. I think having both is helpful because:

- Having both helps show intent. Transform is intended for creating new trees based on old trees. Walk is intended for side effects.
- Having both improves efficiency. If you were to implement walk using transform, you would allocate a duplicate tree and then immediately throw it away.

Reply:
This makes sense, I agree that a separate walk function would be beneficial